packages/shell-api/src/collection.ts

```diff
@@ -1475,6 +1476,231 @@ export default class Collection extends ShellApiWithMongoClass {
     return new Explainable(this._mongo, this, verbosity);
   }
 
+  /**
+   * Running the $collStats stage on sharded timeseries clusters
+   * fails on some versions of MongoDB. SERVER-72686
+   * This function provides the deprecated fallback in those instances.
+   */
+  async _getLegacyCollStats(scale: number) {
+    const result = await this._database._runCommand(
+      {
+        collStats: this._name,
+        scale: scale || 1
+      }
+    );
+
+    if (!result) {
+      throw new MongoshRuntimeError(
+        `Error running collStats command on ${this.getFullName()}`,
+        CommonErrors.CommandFailed
+      );
+    }
+
+    return result;
+  }
+
+  /**
+   * Build a single scaled collection stats result document from the
+   * potentially multiple documents returned from the `$collStats` aggregation
+   * result. We run the aggregation stage with scale 1 and scale it here
+   * in order to accurately scale the summation of stats across various shards.
+   */
+  async _aggregateAndScaleCollStats(collStats: Document[], scale: number) {
+    const result: Document = {
+      ok: 1,
+    };
+
+    const shardStats: {
+      [shardId: string]: Document
+    } = {};
+    const counts: {
+      [fieldName: string]: number;
+    } = {};
+    const indexSizes: {
+      [indexName: string]: number;
+    } = {};
+    const clusterTimeseriesStats: {
+      [statName: string]: number
+    } = {};
+
+    let maxSize = 0;
+    let unscaledCollSize = 0;
+
+    let nindexes = 0;
+    let timeseriesBucketsNs: string | undefined;
+    let timeseriesTotalBucketSize = 0;
+
+    for (const shardResult of collStats) {
+      const shardStorageStats = shardResult.storageStats;
+
+      // We don't know the order that we will encounter the count and size, so we save them
+      // until we've iterated through all the fields before updating unscaledCollSize
+      // Timeseries bucket collection does not provide 'count' or 'avgObjSize'.
+      const countField = shardStorageStats.count;
+      const shardObjCount = (typeof countField !== 'undefined') ? countField : 0;
+
+      for (const fieldName of Object.keys(shardStorageStats)) {
+        if (['ns', 'ok', 'lastExtentSize', 'paddingFactor'].includes(fieldName)) {
+          continue;
+        }
+        if (
+          [
+            'userFlags', 'capped', 'max', 'paddingFactorNote', 'indexDetails', 'wiredTiger'
+          ].includes(fieldName)
+        ) {
+          // Fields that are copied from the first shard only, because they need to
+          // match across shards.
+          if (result[fieldName] === undefined) {
+            result[fieldName] = shardStorageStats[fieldName];
+          }
```

Comment:
(Feel free to ignore) If you want to simplify this a bit:

```suggestion
          result[fieldName] ??= shardStorageStats[fieldName];
```

Reply:
Ah nice! I somehow often forget this operator exists.